---
title: 'Important biological information uncovered in previously unaligned reads from chromatin immunoprecipitation experiments (ChIP-Seq)'
date: 2015-03-01
authors:
  - Wilberforce Zachary Ouma
  - Maria Katherine Mejia-Guerra  
  - Alper Yilmaz
  - Pablo Pareja-Tobes 
  - Wei Li 
  - Andrea I Doseff 
  - Erich Grotewold

publication_types:
  - 2

abstract: |
  Establishing the architecture of gene regulatory networks (GRNs) relies on chromatin immunoprecipitation followed by massively parallel sequencing (ChIP-Seq) methods that provide genome-wide transcription factor binding sites (TFBSs). ChIP-Seq furnishes millions of short reads that, after alignment, describe the genome-wide binding sites of a particular TF. However, in all organisms investigated an average of 40% of reads fail to align to the corresponding genome, with some datasets having as much as 80% of reads failing to align. We describe here the provenance of previously unaligned reads in ChIP-Seq experiments from animals and plants. We show that a substantial portion corresponds to sequences of bacterial and metazoan origin, irrespective of the ChIP-Seq chromatin source. Unforeseen was the finding that 30%-40% of unaligned reads were actually alignable. To validate these observations, we investigated the characteristics of the previously unaligned reads corresponding to TAL1, a human TF involved in lineage specification of hemopoietic cells. We show that, while unmapped ChIP-Seq read datasets contain foreign DNA sequences, additional TFBSs can be identified from the previously unaligned ChIP-Seq reads. Our results indicate that the re-evaluation of previously unaligned reads from ChIP-Seq experiments will significantly contribute to TF target identification and determination of emerging properties of GRNs.

featured: false
publication: "*Scientific Reports*"
tags:
  - Gene regulatory networks
  - Bioinformatics
url_pdf: "https://www.ncbi.nlm.nih.gov/pubmed/25727450"
doi: 10.1038/srep08635
---
